File: src/java/foundgine-providers/src/main/java/com/foundgine/providers/storage/sql/retrieval/PostgresRetrievalOptions.java
package com.foundgine.providers.storage.sql.retrieval;
public record PostgresRetrievalOptions(boolean enablePgTrgm, boolean enableFullText, boolean enablePgSearch, boolean enableApacheAge, String fullTextConfiguration, String ageGraphName) {
 public PostgresRetrievalOptions(){this(true,true,false,false,"english","foundgine");}
}

File: src/csharp/Foundgine.Providers/Storage/Sql/Retrieval/PostgresRetrievalOptions.cs
namespace Foundgine.Providers.Storage.Sql.Retrieval;

/// <summary>PostgreSQL retrieval capabilities used to ground semantic references.</summary>
public sealed record PostgresRetrievalOptions(
    bool EnablePgTrgm = true,
    bool EnableFullText = true,
    bool EnablePgSearch = false,
    bool EnableApacheAge = false,
    string FullTextConfiguration = "english",
    string AgeGraphName = "foundgine");